diff -ruN ffmpeg/libavformat/mpegts.c FFmpeg_SCTE35/libavformat/mpegts.c
--- ffmpeg/libavformat/mpegts.c	2025-02-27 09:52:45.461300528 -0500
+++ FFmpeg_SCTE35/libavformat/mpegts.c	2025-02-27 09:45:49.009297761 -0500
@@ -877,6 +877,7 @@
     { MKTAG('I', 'D', '3', ' '), AVMEDIA_TYPE_DATA,  AV_CODEC_ID_TIMED_ID3 },
     { MKTAG('V', 'C', '-', '1'), AVMEDIA_TYPE_VIDEO, AV_CODEC_ID_VC1   },
     { MKTAG('O', 'p', 'u', 's'), AVMEDIA_TYPE_AUDIO, AV_CODEC_ID_OPUS  },
+    { MKTAG('C', 'U', 'E', 'I'), AVMEDIA_TYPE_DATA,  AV_CODEC_ID_SCTE_35 },
     { 0 },
 };
 
diff -ruN ffmpeg/libavformat/mpegtsenc.c FFmpeg_SCTE35/libavformat/mpegtsenc.c
--- ffmpeg/libavformat/mpegtsenc.c	2025-02-27 09:52:45.461300528 -0500
+++ FFmpeg_SCTE35/libavformat/mpegtsenc.c	2025-02-27 09:45:49.009297761 -0500
@@ -443,6 +443,10 @@
             stream_type = STREAM_TYPE_PRIVATE_DATA;
         }
         break;
+    case AV_CODEC_ID_SCTE_35:
+    	stream_type = STREAM_TYPE_SCTE_DATA_SCTE_35;
+        st->codecpar->codec_type = AVMEDIA_TYPE_DATA;
+    	break;
     default:
         av_log_once(s, AV_LOG_WARNING, AV_LOG_DEBUG, &ts_st->data_st_warning,
                     "Stream %d, codec %s, is muxed as a private data stream "
@@ -813,7 +817,9 @@
             }
             break;
         case AVMEDIA_TYPE_DATA:
-            if (codec_id == AV_CODEC_ID_SMPTE_KLV) {
+           if (codec_id == AV_CODEC_ID_SCTE_35) {
+                put_registration_descriptor(&q, MKTAG('C', 'U', 'E', 'I'));        
+           } else if (codec_id == AV_CODEC_ID_SMPTE_KLV) {
                 put_registration_descriptor(&q, MKTAG('K', 'L', 'V', 'A'));
             } else if (codec_id == AV_CODEC_ID_SMPTE_2038) {
                 put_registration_descriptor(&q, MKTAG('V', 'A', 'N', 'C'));
